spec/factories/ext_management_system.rb

```diff
@@ -0,0 +1,34 @@
+FactoryGirl.define do
+  factory :ems_nuage_with_vcr_authentication, :parent => :ems_nuage_network do
+    zone do
+      _guid, _server, zone = EvmSpecHelper.create_guid_miq_server_zone
+      zone
+    end
+
+    after(:build) do |ems|
+      ems.hostname = Rails.application.secrets.nuage_network.try(:[], 'host') || 'nuagenetworkhost'
+    end
+
+    after(:create) do |ems|
+      userid   = Rails.application.secrets.nuage_network.try(:[], 'userid') || 'NUAGE_USER_ID'
+      password = Rails.application.secrets.nuage_network.try(:[], 'password') || 'NUAGE_PASSWORD'
+
+      # Ensure that VCR will obfuscate the basic auth
+      VCR.configure do |c|
```

Comment:
@gberginc I think configuring "VCR recording rules" inside "database object factory" makes no sense no matter what. The factory is only meant to instantiate Rails objects for us, nothing more.

As far as I can understand, the
```ruby
VCR.configure do |config|
  ...
end
```
snippet only needs to be executed once and the `NUAGE_NETWORK_AUTHORIZATION` will be replaced for every subsequent casette, right? In other words: do we really need to configure VCR before each test? I think this is not the case, therefore I suggest we move the VCR code from `spec/factories/ext_management_system.rb` to `spec/spec_helper.rb` where we merge it with current VCR setting. Would it work this way?

If configuring it in `spec/spec_helper.rb` for some reason doesn't work, then I'd put the VCR configuration code into corresponding spec as VMware does (i.e. the second bullet from your comment. THB, I don't see the difference between the two bullets that you provided - links are the same).

In both cases we will read userid and password from yaml once again, but I don't think this is a problem to worry about. That being said, I'm not sure why we even want to read it from yaml at first place - wouldn't we be better off hard-coding it? I may be missing something, but if we only use yaml for that rare occasion when VCR cassete is being recorded, then it would probably be easier to temporarily provide actual username/password into variable directly...